repositories/src/main/scala/cromwell/pipeline/datastorage/dto/ProjectSearch.scala

```diff
@@ -0,0 +1,103 @@
+package cromwell.pipeline.datastorage.dto
+
+import cromwell.pipeline.utils.json.AdtJsonFormatter
+import play.api.libs.functional.syntax.toInvariantFunctorOps
+import play.api.libs.json.JsonConfiguration.Aux
+import play.api.libs.json._
+
+sealed trait ProjectSearchFilter
+object ProjectSearchFilter {
+  implicit lazy val jsonConfig: Aux[Json.MacroOptions] = JsonConfiguration(
+    discriminator = "type",
+    typeNaming = JsonNaming { fullName =>
+      val jsName = fullName.split("\\.").lastOption.fold("")(_.toLowerCase)
+      jsName match {
+        case "byname"   => "name"
+        case "byconfig" => "configurations"
+        case "byfiles"  => "files"
+        case _          => jsName
+      }
+    }
+  )
+  implicit lazy val projectSearchFilterFormat: OFormat[ProjectSearchFilter] = Json.format[ProjectSearchFilter]
+}
+
+case object All extends ProjectSearchFilter {
+  implicit lazy val allFormat: OFormat[All.type] = AdtJsonFormatter.objectFormat(All)
```

Comment:
If you've already found this class (`AdtJsonFormatter.objectFormat`), why not use it for `ProjectSearchFilter` and not only for one of its cases? I imagine it will make code cleaner

Reply:
Hmm, sorry, I've approved MR thinking that there is only one minor thing, but I see that this comment is not resolved yet

Reply:
Refactored with `AdtJsonFormatter`

Reply:
Made a private method to generalize formatter creation for `Or` and `And` filters. Couldn't make `projectSearchFilter: OFormat` to be parameter though.